tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -219,6 +227,197 @@ def set_system_purpose_test(self):
+        subscription_request = {
+            "type": get_variant(Str, SUBSCRIPTION_REQUEST_TYPE_ORG_KEY),
+            "organization": get_variant(Str, "123456789"),
+            "activation-keys": get_variant(List[Str], ["key_foo", "key_bar", "key_baz"])
+        }
+        expected_subscription_request = {
+            "type": get_variant(Str, SUBSCRIPTION_REQUEST_TYPE_ORG_KEY),
+            "organization": get_variant(Str, "123456789"),
+            "account-username": get_variant(Str, ""),
+            "server-hostname": get_variant(Str, ""),
+            "rhsm-baseurl": get_variant(Str, ""),
+            "server-proxy-hostname": get_variant(Str, ""),
+            "server-proxy-port": get_variant(Int, -1),
+            "server-proxy-user": get_variant(Str, ""),
+            "account-password": get_variant(Str, ""),
+            "activation-keys": get_variant(List[Str], []),
+            "server-proxy-password": get_variant(Str, ""),
+            "account-password-set": get_variant(Bool, False),
+            "activation-keys-set": get_variant(Bool, True),
+            "server-proxy-password-set": get_variant(Bool, False),
+        }
+
+        self._test_dbus_property(
+          "SubscriptionRequest",
+          subscription_request,
+          expected_subscription_request
+        )
+
+    def set_subscription_request_proxy_test(self):
+        """Test if setting HTTP proxy in subscription request from DBUS works correctly."""
+        subscription_request = {
+            "server-proxy-hostname": get_variant(Str, "proxy.foo.bar"),
+            "server-proxy-port": get_variant(Int, 9001),
+            "server-proxy-user": get_variant(Str, "foo_proxy_user"),
+            "server-proxy-password": get_variant(Str, "foo_proxy_password"),
+        }
+        expected_subscription_request = {
+            "type": get_variant(Str, SUBSCRIPTION_REQUEST_TYPE_USERNAME_PASSWORD),
+            "organization": get_variant(Str, ""),
+            "account-username": get_variant(Str, ""),
+            "server-hostname": get_variant(Str, ""),
+            "rhsm-baseurl": get_variant(Str, ""),
+            "server-proxy-hostname": get_variant(Str, "proxy.foo.bar"),
+            "server-proxy-port": get_variant(Int, 9001),
+            "server-proxy-user": get_variant(Str, "foo_proxy_user"),
+            "account-password": get_variant(Str, ""),
+            "activation-keys": get_variant(List[Str], []),
+            "server-proxy-password": get_variant(Str, ""),
+            "account-password-set": get_variant(Bool, False),
+            "activation-keys-set": get_variant(Bool, False),
+            "server-proxy-password-set": get_variant(Bool, True),
+        }
+
+        self._test_dbus_property(
+          "SubscriptionRequest",
+          subscription_request,
+          expected_subscription_request
+        )
+
+    def set_subscription_request_custom_urls_test(self):
```

Comment:
I think you should have at least one test using those data structures.

Reply:
I'm not sure what you mean by "use" ? The tests check that data in the structures can be set & read out. Actual use will come in subsequent PRs when tasks and the GUI use the SubscriptionReqeust structure to process and exchange data.

Reply:
I mean create `SubscriptionRequest` and `SystemPurposeData` here and use them in the interface.

Reply:
Oh, right - yeah, I can add a test like that. :)